shared/chat/conversation/attachment-input/container.js

```diff
@@ -0,0 +1,44 @@
+// @flow
+import RenderAttachmentInput from './'
+import {connect} from 'react-redux'
+import {navigateUp} from '../../../actions/route-tree'
+
+import type {RouteProps} from '../../../route-tree/render-route'
+import type {TypedState} from '../../../constants/reducer'
+import type {AttachmentInput, SelectAttachment} from '../../../constants/chat'
+
+type AttachmentInputRouteProps = RouteProps<{
+  input: AttachmentInput,
+}, {}>
+type OwnProps = AttachmentInputRouteProps & {
+  title: string,
+}
+
+export default connect(
+  (state: TypedState, {routeProps, ...ownProps}: OwnProps) => {
+    const {input} = routeProps
+    return {
+      ...ownProps,
+      input,
+      title: input.title,
+    }
+  },
+  (dispatch: Dispatch) => ({
+    onClose: () => dispatch(navigateUp()),
+    onSelect: (input: AttachmentInput) => {
+      dispatch(navigateUp())
+      dispatch(({payload: {input}, type: 'chat:selectAttachment'}: SelectAttachment))
+    },
+  }),
+  (stateProps, dispatchProps) => {
+    const {input} = stateProps
+    return {
+      ...stateProps,
+      ...dispatchProps,
+      onSelect: (title: string) => {
+        input.title = title
```

Comment:
@chromakode  hmmm, will this actually work correctly? like updating this way will cause it to re-render and get the props? we don't need some kind of setState for routeState?